contracts/strategies/LPOptimiserStrategy.sol

```diff
@@ -12,94 +12,121 @@ contract LPOptimiserStrategy is DefaultAccessControl, ILpCallback {
     using SafeERC20 for IERC20;
     using PRBMathUD60x18 for uint256;
 
+    // VAULT PARAMETERS
+    struct VaultParams {
+        int256 sigmaWad; // standard deviation parameter in wad 10^18
+        int256 maxPossibleLowerBoundWad; // Maximum Possible Fixed Rate Lower bounds when initiating a rebalance
+        uint256 proximityWad; // closeness parameter in wad 10^18
+        uint256 weight; // weight parameter that decides how many funds are going to this vault
+    }
+
     // IMMUTABLES
     address[] public _tokens;
     IERC20Vault public immutable _erc20Vault;
 
     // INTERNAL STATE
-    IVoltzVault internal _vault;
-    IMarginEngine internal _marginEngine;
-    IPeriphery internal _periphery;
-    IVAMM internal _vamm;
-
-    // MUTABLE PARAMS
-    uint256 internal _sigmaWad; // y (standard deviation parameter in wad 10^18)
-    int256 internal _maxPossibleLowerBoundWad; // Maximum Possible Fixed Rate Lower bounds when initiating a rebalance
-    uint256 internal _proximityWad; // x (closeness parameter in wad 10^18)
+    IVoltzVault[] internal _vaults;
+    VaultParams[] internal _vaultParams;
+    uint256 _totalWeight; // sum of all vault weights
 
     // CONSTANTS
-    uint256 internal constant MINIMUM_FIXED_RATE = 1e16;
+    int256 internal constant MINIMUM_FIXED_RATE = 1e16;
     uint256 internal constant LOG_BASE = 1000100000000000000;
 
     // GETTERS AND SETTERS
-    function setSigmaWad(uint256 sigmaWad) public {
-        _requireAdmin();
-        _sigmaWad = sigmaWad;
-    }
-
-    function setMaxPossibleLowerBound(int256 maxPossibleLowerBoundWad) public {
-        _requireAdmin();
-        _maxPossibleLowerBoundWad = maxPossibleLowerBoundWad;
-    }
-
-    function setProximityWad(uint256 proximityWad) public {
-        _requireAdmin();
 
-        _proximityWad = proximityWad;
+    /// @notice Get the addresses of all vaults
+    function getVaults() public view returns (IVoltzVault[] memory) {
+        return _vaults;
     }
 
-    function getSigmaWad() public view returns (uint256) {
-        return _sigmaWad;
+    /// @notice Get the parameters of a vault 
+    /// @param index The index of the vault in _vaults
+    function getVaultParams(uint256 index) public view returns (VaultParams memory) {
+        return _vaultParams[index];
     }
 
-    function getMaxPossibleLowerBound() public view returns (int256) {
-        return _maxPossibleLowerBoundWad;
-    }
+    /// @notice Set the parameters of a vault
+    function setVaultParams(uint256 index, VaultParams memory vaultParams_) external {
+        _requireAdmin();
+        require(index < _vaults.length, ExceptionsLibrary.INVALID_STATE);
 
-    function getProximityWad() public view returns (uint256) {
-        return _proximityWad;
+        uint256 previousWeight = _vaultParams[index].weight;
```

Comment:
Can we get rid of this var if we update _vaultParams after we update _totalWeight?

Reply:
Done